byzcoin/service.go

```diff
@@ -2305,6 +2311,16 @@ func (s *Service) startChain(genesisID skipchain.SkipBlockID) error {
 		return nil
 	}
 
+	// before doing anything, verify that byzcoin is consistent
+	st, err := s.getStateTrie(genesisID)
+	if err != nil {
+		return err
+	}
+	if err := s.fixInconsistencyIfAny(genesisID, st); err != nil {
```

Comment:
I might be missing something but if you only repair the inconsistencies when booting the conode, that means that until you restart the conode, one inconsistency can be propagated to client requests / catch up / whatever ?

Reply:
Yes, this is the intended behaviour. The issue that this PR tries to solve is: how to make sure the trie and skipblocks are consistent after a crash or power failure. Since we store the skipblock and then the trie one of the other, the only time that the database becomes inconsistent internally is when a crash happens in between these two operations. The inconsistency is only visible after a restart, that's why we do the repair at startup. https://github.com/dedis/cothority/issues/1982#issuecomment-507201045 explains it in another way